tests/test_subclass_contracts.py

```diff
@@ -128,3 +171,35 @@ def test_change_teamlead_team_invalid(teamlead):
         teamlead.team = ''
     msg = str(excinfo.value)
     assert 'len(self.team) > 0' in msg
+
+
+def test_teacher_rep_violatable(pe_bio_teacher):
```

Comment:
I think this test and `test_method_violates_class_invariant` are redundant with the ones using the `Developer` class.

Reply:
I'm making a case that I want to keep one of the two as a test to check that the class' invariant is written properly and actually works so it isn't being skipped and throwing false positives.

Reply:
I see what you're saying, but I do not think this is important enough to justify the additional code. However, instead of creating a new subclass you can reuse the existing subclass (`Developer`) and just create a new method and test with that.

That said, next time I ask you to do something, you should do it even if you want to argue against it. That way I can respond and merge in your pull request at once, rather than waiting another cycle.

Reply:
Understood. I'll make sure to do that in the future